### PR TITLE
add installation login tests

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/roadmap/issues/798
---

As the creator of a pull request, please consider:

- [ ] Describe the goal you are trying to accomplish here, above the line.
- [ ] Add a user-friendly description of your change to `CHANGELOG.md`.
- [ ] Provide a link to the issue you are solving or working towards, if available.
- [ ] Request SIG UX for review. They care about almost all user-facing changes.
- [ ] Update the public [kubectl-gs documentation](https://docs.giantswarm.io/ui-api/kubectl-gs/) to reflect the changes here.
- [ ] add the `breaking-change` label to the PR if the change you are making changes the existing behaviour. Examples: removal of a flag, removal of a command, change of a default value. (Such changes should be released with a **major version** bump.)

Feel free to remove this checklist when done.

### CODE REVIEW

File: cmd/login/mc.go [hunk 1]
```diff
@@ -101,6 +101,41 @@ func (r *runner) loginWithURL(ctx context.Context, path string, firstLogin bool,
 	if installation.GetUrlType(path) == installation.UrlTypeHappa {
 		fmt.Fprint(r.stdout, color.YellowString("Note: deriving Management API URL from web UI URL: %s\n", i.K8sApiURL))
 	}
+	err = r.loginWithInstallation(ctx, tokenOverride, i)
```
Comment (anvddriesch) [inline cmd/login/mc.go:104]: some refactoring. We move the actual login process to a separate function.

File: cmd/login/runner_test.go [hunk 2]
```diff
@@ -114,6 +118,103 @@ func TestLogin(t *testing.T) {
 	}
 }
 
+func TestMCLoginWithInstallation(t *testing.T) {
+	testCases := []struct {
+		name        string
+		startConfig *clientcmdapi.Config
+		flags       *flag
+		expectError *microerror.Error
```
Comment (anvddriesch) [inline cmd/login/runner_test.go:126]: We can add more cases here and also an actual expected kubeconfig outcome

File: cmd/login/runner_test.go [hunk 3]
```diff
@@ -114,6 +118,103 @@ func TestLogin(t *testing.T) {
...
+			}
+			originConfig, err := r.k8sConfigAccess.GetStartingConfig()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ctx := context.TODO()
+			r.setLoginOptions(ctx, []string{"codename"})
+
+			err = r.loginWithInstallation(ctx, "token", CreateTestInstallation())
```
Comment (anvddriesch) [inline cmd/login/runner_test.go:189]: We add the token override here to avoid the OIDC process. For that we need a separate test.

**Comment by marians** [comment]
`coverage: 24.0% of statements` 🚀

**Review by marians — APPROVED** [review]
I did a very shallow check

**Pull request merged**